### PR TITLE
StatusIcon Refactoring and Generalization

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

- VMIconStatus -> IconStatus
- other small fixes

### CODE REVIEW

**Comment by glekner** [comment]
/lgtm

next up move the template svg icons from kubevirt plugin to here

File: src/components/icons/IconStatus/IconStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  ExclamationCircleIcon,
+  InProgressIcon,
+  OffIcon,
+  PausedIcon,
+  SyncAltIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import { global_danger_color_100 as dangerColor } from '@patternfly/react-tokens/dist/js/global_danger_color_100';
+
+interface IconProps {
```
Comment (metalice) [inline src/components/icons/IconStatus/IconStatus.tsx:13]: consider change to type :)

File: src/components/icons/IconStatus/IconStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  title: string;
+}
+
+function RedExclamationCircleIcon({ title }: IconProps): React.ReactElement {
```
Comment (metalice) [inline src/components/icons/IconStatus/IconStatus.tsx:17]: please use () => {} , also, React.FC<props> = ()=>{}

File: src/components/icons/IconStatus/IconStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  return <ExclamationCircleIcon color={dangerColor.value} title={title} />;
+}
+
+function SpinningInProgressIcon({ title }: IconProps): React.ReactElement {
+  return <InProgressIcon title={title} className="fa-spin" />;
```
Comment (metalice) [inline src/components/icons/IconStatus/IconStatus.tsx:22]: consider fa-spin as a const in utils file

File: src/components/icons/IconStatus/IconStatus.tsx (hunk 1 above)
Comment (metalice) [reply src/components/icons/IconStatus/IconStatus.tsx:13]: what about data-test ? :)

File: src/components/icons/IconStatus/IconStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+   * - ImagePullBackOff
+   * - ErrorPvcNotFound
+   * - ErrorDataVolumeNotFound
+   * - DataVolumeError
+   * - WaitingForVolumeBinding
+   * */
+  status: string;
+};
+
+export function IconStatus({ status }: IconStatusProps) {
```
Comment (metalice) [inline src/components/icons/IconStatus/IconStatus.tsx:148]: same comments as above :)

File: styleguide/components/Type.tsx [hunk 5]
```diff
@@ -10,16 +10,15 @@ export const styles = ({ fontFamily, fontSize, color }: Theme) => ({
...
 interface TypeProps extends JssInjectedProps {
   children: React.ReactNode;
 }
 
 function getGithubTypeHref(type: string): string {
-  return `https://github.com/kubevirt-ui/kubevirt-api/blob/main/kubevirt/models/${type.replace(
-    /\[\]/,
-    '',
-  )}.ts`;
+  return `https://github.com/kubevirt-ui/kubevirt-api/blob/main/${
```
Comment (metalice) [inline styleguide/components/Type.tsx:19]: should https://github.com/kubevirt-ui/kubevirt-api/blob/main/ be used as CONST ?

File: styleguide/components/Type.tsx [hunk 6]
```diff
@@ -10,16 +10,15 @@ export const styles = ({ fontFamily, fontSize, color }: Theme) => ({
...
+    KUBEVIRT_CONSTS.KUBEVIRT_MODEL_PATH
+  }/${type.replace(/\[\]/, '')}.ts`;
 }
 
 function TypeRenderer({ classes, children }: TypeProps) {
```
Comment (metalice) [inline styleguide/components/Type.tsx:24]:
```suggestion
const TypeRenderer: React.FC<TypeProps> = ({ classes, children }) => {
  const isWrappedWithLink = typeof children === 'string' && KUBEVIRT_TYPES.includes(children.replace(/\[\]/, ''))

  return <span className={classes.type}>{
  isWrappedWithLink ? 
  <a href={getGithubTypeHref(children)} target="_blank" rel="noreferrer">
        {children}
      </a> : 
      children}</span>;
}
```

Comment (upalatucci) [reply styleguide/components/Type.tsx]: about functional components, there is a rule of Deepsource that does not like them: https://deepsource.io/gh/kubevirt-ui/kubevirt-components/run/465793fe-269b-427f-a88d-a2839956f249/javascript/JS-0399/
@glekner .
For this reason, they are all declared in the old fashion way :D

File: src/components/icons/IconStatus/IconStatus.tsx (hunk 1 above)
Comment (upalatucci) [reply src/components/icons/IconStatus/IconStatus.tsx]: What do you mean with data-test? Should we add a prop to provide data-test attribute?

File: styleguide/components/Type.tsx (hunk 6 above)
Comment (upalatucci) [reply styleguide/components/Type.tsx]: disabled!

File: src/components/icons/IconStatus/IconStatus.tsx (hunk 1 above)
Comment (upalatucci) [reply src/components/icons/IconStatus/IconStatus.tsx]: Good idea!

**Comment by upalatucci** [comment]
@metalice thanks. I resolved all the suggestions. What do you think about the data-test solution that I implemented?

File: src/components/icons/IconStatus/IconStatus.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  SyncAltIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import { global_danger_color_100 as dangerColor } from '@patternfly/react-tokens/dist/js/global_danger_color_100';
+
+import { faSpin } from './utils';
+
+type IconProps = {
+  title: string;
+  'data-test'?: string;
```
Comment (glekner) [inline src/components/icons/IconStatus/IconStatus.tsx:17]: replace all with `data-test-id`
this is console convention cc @metalice @yaacov

**Comment by upalatucci** [comment]
@yaacov @glekner  can you please review this?

**Pull request closed without merging**